src/WebJobs.Script.WebHost/Program.cs

```diff
@@ -36,9 +45,16 @@ public static IWebHostBuilder CreateWebHostBuilder(string[] args = null)
                     {
                         config.Sources.Remove(envVarsSource);
                     }
-                    envVarsSource = new ScriptEnvironmentVariablesConfigurationSource();
-                    config.Sources.Add(envVarsSource);
+
+                    config.Add(new ScriptEnvironmentVariablesConfigurationSource());
+
+                    config.Add(new WebScriptHostConfigurationSource
+                    {
+                        IsAppServiceEnvironment = SystemEnvironment.Instance.IsAppServiceEnvironment(),
```

Comment:
Why is this needed? I'd think this logic should be internalized somehow. The config steps for a simple host like this should just be a few lines.